go/stats/prombackend/collectors.go

```diff
@@ -0,0 +1,99 @@
+package prombackend
+
+import (
+	"strings"
+
+	prom "github.com/prometheus/client_golang/prometheus"
+	"vitess.io/vitess/go/stats"
+)
+
+// TODO(zmagg): Consider using NewConstMetric and handling error here instead of panic-ing
+
+type multiCountersCollector struct {
+	multiCounters map[string]*prom.Desc // More information about the metrics is keyed by the same string and in pullMetrics
+}
+
+func NewMultiCountersCollector(multiCounters map[string]*prom.Desc) prom.Collector {
```

Comment:
None of these NewXYZCollector functions should be exposed as part of the public API.

I think you should either make them `newXYZCollector` or just inline the struct creation into `prombackend.NewMultiCounter`

Reply:
This is great feedback, this is a lot clearer inlined.